plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -661,14 +694,23 @@ public SearchResponse beginSearch(String index, int shard, QueryBuilder query, O
         });
         documentFields.forEach(sourceBuilder::docValueField);
 
-        LOG.debug("Begin search: %s:%s, query: %s", index, shard, sourceBuilder);
-
         SearchRequest request = new SearchRequest(index)
                 .searchType(QUERY_THEN_FETCH)
-                .preference("_shards:" + shard)
-                .scroll(new TimeValue(scrollTimeout.toMillis()))
                 .source(sourceBuilder);
 
+        // apply aggregation to sourceBuilder if the query includes aggregation
+        // also must not use _shards and scroll in this case
+        if (aggregations.isPresent()) {
+            aggregations.get().forEach(sourceBuilder::aggregation);
+            sourceBuilder.size(0);
```

Comment:
Move this to the block at the top where it makes the decision about `sourceBuilder.size()`. You'll need to add an extra branch to the `if-else` that checks for `aggregations.isPresent()`